### PR TITLE
Add support for match statement

### LINKED ISSUE #215: AttributeError: No defined handler for node of type Match
Here are the relevant error messages, triggered by `astor.to_source(tree)`

```
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 63, in to_source
    generator.visit(node)
  File "C:\py\Python311\Lib\site-packages\astor\node_util.py", line 143, in visit
    return visitor(node)
           ^^^^^^^^^^^^^
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 335, in visit_FunctionDef
    self.body(node.body)
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 226, in body
    self.write(*statements)
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 178, in write
    visit(item)
  File "C:\py\Python311\Lib\site-packages\astor\node_util.py", line 143, in visit
    return visitor(node)
           ^^^^^^^^^^^^^
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 397, in visit_While
    self.body_or_else(node)
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 235, in body_or_else
    self.body(node.body)
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 226, in body
    self.write(*statements)
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 178, in write
    visit(item)
  File "C:\py\Python311\Lib\site-packages\astor\node_util.py", line 143, in visit
    return visitor(node)
           ^^^^^^^^^^^^^
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 388, in visit_For
    self.body_or_else(node)
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 235, in body_or_else
    self.body(node.body)
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 226, in body
    self.write(*statements)
  File "C:\py\Python311\Lib\site-packages\astor\code_gen.py", line 178, in write
    visit(item)
  File "C:\py\Python311\Lib\site-packages\astor\node_util.py", line 143, in visit
    return visitor(node)
           ^^^^^^^^^^^^^
  File "C:\py\Python311\Lib\site-packages\astor\node_util.py", line 137, in abort_visit
    raise AttributeError(msg % node.__class__.__name__)
AttributeError: No defined handler for node of type Match
```

### PR DESCRIPTION
**Pull request by askurihin**
Add match statement support, supersedes #195.
Fixes #215.

Please let me know if anything should be added/adjusted.

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -8,6 +8,8 @@ Release Notes
 Bug fixes
 ~~~~~~~~~
 
+* Add support for Structural Pattern Matching
```
Comment (berkerpeksag) [inline docs/changelog.rst:11]: Nitpick: Can we add a link to the PEP?

Comment (berkerpeksag) [reply docs/changelog.rst:11]: We also need to add "(Reported by avelican in Issue 215 and contributed by Skurikhin Alexandr in PR 219.)"

Comment (berkerpeksag) [reply docs/changelog.rst:11]: Oh, and this needs to be into a new section called "New Features" :) See 0.8.0 for an example.

File: tests/test_code_gen.py [hunk 2]
```diff
@@ -641,6 +641,300 @@ def test_compile_types(self):
             if code.strip() != dsttxt.strip():
                 self.assertEqual('(%s)' % code.strip(), dsttxt.strip())
 
+    @unittest.skipUnless(sys.version_info >= (3, 10, 0),
+                         "match statement introduced in Python 3.10")
+
```
Comment (berkerpeksag) [inline tests/test_code_gen.py:646]: Nitpick: Can we remove the extra newline here and below?

**Review by berkerpeksag — CHANGES_REQUESTED** [review]
Thank you for the PR. I need to read the PEP again, but overall this looks pretty good after fixing the nitpicks I mentioned in the review. I think we can get it in and then improve it later if needed.

File: tests/test_code_gen.py (hunk 2 above)
Comment (askurihin) [reply tests/test_code_gen.py]: Fixed, thanks!

File: docs/changelog.rst (hunk 1 above)
Comment (askurihin) [reply docs/changelog.rst]: All done, thank you

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Thank you very much!

File: tests/test_code_gen.py (hunk 2 above)
Comment (berkerpeksag) [reply tests/test_code_gen.py]: No worries! I wasn't super clear anyway :)

**Comment by agomez-etsy** [comment]
Hi @berkerpeksag. I see this change was merged early this year. Are there any plans to release a new version of astor with this fix? I have a package that depends on this, hence my question 🙌

**Comment by berkerpeksag** [comment]
Apologies for not releasing this sooner! I'll get to it this weekend and ping you here.

**Comment by agomez-etsy** [comment]
Thanks @berkerpeksag 🙇

**Comment by agomez-etsy** [comment]
Hi @berkerpeksag ! I was wondering if you can release the new version in the upcoming days. No rush if not, just wanted to check

**Pull request merged**